tensorflow/tsl/platform/default/env.cc

```diff
@@ -76,7 +76,16 @@ class PThread : public Thread {
     pthread_attr_destroy(&attributes);
   }
 
-  ~PThread() override { pthread_join(thread_, nullptr); }
+  ~PThread() override {
+    if (!detached_) {
+      pthread_join(thread_, nullptr);
+    }
+  }
+
+  void detach() override {
```

Comment:
Could you write a docstring and maybe a link to some standard reference on how detached threads should behave? Is it possible to test this?

Reply:
Description added